### ISSUE TITLE
if a repo name changes, cookbook cannot delete/ignore the extra repo name

### ISSUE THREAD

**Issue by joshoohaah** [kemra102/yumserver-cookbook#7]
Stumbled across a new issue on 2.0.2 where if i changed the name of the repo name i'm mirroring it has issues doing `delete`.
See error message.
```


    ================================================================================
    Error executing action `delete` on resource 'zap_directory[/etc/reposync.repos.d/]'
    ================================================================================

    Chef::Exceptions::ValidationFailed
    ----------------------------------
    yumserver_mirror[/etc/reposync.repos.d/epel7.repo] (dynamically defined) had an error: Chef::Exceptions::ValidationFailed: repo_name is required

    Cookbook Trace:
    ---------------
    /var/chef/cache/cookbooks/yumserver/resources/mirror.rb:19:in `path'
    /var/chef/cache/cookbooks/yumserver/resources/mirror.rb:66:in `block in class_from_file'
    /var/chef/cache/cookbooks/zap/libraries/default.rb:149:in `block (2 levels) in iterate'
    /var/chef/cache/cookbooks/zap/libraries/default.rb:145:in `each'
    /var/chef/cache/cookbooks/zap/libraries/default.rb:145:in `block in iterate'
    /var/chef/cache/cookbooks/zap/libraries/default.rb:144:in `iterate'
    /var/chef/cache/cookbooks/zap/libraries/default.rb:110:in `action_delete'

    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/yumserver/recipes/default.rb

     23: zap_directory '/etc/reposync.repos.d/' do
     24:   klass [Chef::Resource::YumserverMirror, Chef::Resource::YumserverRsyncMirror]
     25:   immediately true
     26: end
     27:

    Compiled Resource:
    ------------------
    # Declared in /var/chef/cache/cookbooks/yumserver/recipes/default.rb:23:in `from_file'

    zap_directory("/etc/reposync.repos.d/") do
      provider Chef::Provider::ZapDirectory
      action :delete
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      pattern "*"
      filter #<Proc:0x000000046d9b98@/var/chef/cache/cookbooks/zap/libraries/default.rb:33 (lambda)>
      klass [LWRP resource yumserver_mirror from cookbook yumserver, LWRP resource yumserver_rsync_mirror from cookbook yumserver]
      declared_type :zap_directory
      cookbook_name "yumserver"
      recipe_name "default"
      immediately true
    end
```

**Comment by joshoohaah** [comment]
please disregard. i think this issue is due to me not setting the new attribute `default['yumserver']['basepath']`

**Comment by joshoohaah** [comment]
I think the `use_repo` setting is causing me this consternation!  i think some of the sub-packages `xz` was pulling from wrong epel repo.  i'll set `use_repo` to  `false` and see if this fixes the issue.

**Comment by joshoohaah** [comment]
YES. :boom:   for historical purposes to fix this I was on centos 6.6.  i was trying to host/reposync a centos 7  epel7 repo.   having `use_repo` set to true then made this package `pyliblzma` install the wrong rpm (rpm meant for 7, not 6).
cleaning out the epel packages, then installing the correct epel repo, and doing `yum install pyliblzma` allowed reposync to work properly again.
